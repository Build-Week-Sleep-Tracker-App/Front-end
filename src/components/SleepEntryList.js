import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import * as actionCreators from '../actions/actionCreators';

const SleepEntryList = ({ user, editSleepEntry, deleteSleepEntry }) => {
	return (
		<div className="sleepEntryList">
			{
				user.sleepData.map(entry => <div key={entry.id} className="sleepEntry">
					<div><b>Start:</b> {moment(entry.start).format('HH:mm DD/MM/YYYY')}</div>
					<div><b>End:</b> {moment(entry.end).format('HH:mm DD/MM/YYYY')}</div>
					<div><b>Mood levels:</b></div>
					<div><b>Before sleep</b> {entry.bed_t_tracking}</div>
					<div><b>After waking up</b> {entry.work_t_tracking}</div>
					<div><b>During the day</b> {entry.average_rating}</div>
				</div>)
			}
			
		</div>
	)
}

export default connect(state => state, actionCreators)(SleepEntryList);
