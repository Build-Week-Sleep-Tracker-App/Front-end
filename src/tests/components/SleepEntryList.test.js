import React from 'react';
import { shallow } from '../../enzyme';
import configureStore from 'redux-mock-store';

import SleepEntryList from '../../components/SleepEntryList';

const mockStore = configureStore();
const initialState = {
  login: {
    isLoggingIn: false,
    isLoggedIn: true,
    userId: null
  }
};
const store = mockStore(initialState);
const wrapper = shallow(<SleepEntryList store={store} />);

describe('Sleep entry list component', () => {
  test('should render entry list correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
