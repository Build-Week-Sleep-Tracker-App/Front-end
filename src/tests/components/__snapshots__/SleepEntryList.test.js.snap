// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sleep entry list component should render entry list correctly 1`] = `ShallowWrapper {}`;
